docs/common/craft-parts/reference/step_execution_environment.rst

```diff
@@ -6,27 +6,141 @@ Step execution environment
-- ``CRAFT_PRIME``: The path to the final primed payload directory
-  after the ``PRIME`` step.
+.. list-table::
+  :header-rows: 1
+
+  * - Variable Name
+    - Description
+  * - ``CRAFT_ARCH_TRIPLET_BUILD_FOR``
+    - The architecture triplet of the build target. For example:
+
+      - ``x86_64-linux-gnu``
+      - ``riscv64-linux-gnu``
+  * - ``CRAFT_ARCH_TRIPLET_BUILD_ON``
+    - The architecture triplet of the host running the build. For example:
+
+      - ``x86_64-linux-gnu``
+      - ``riscv64-linux-gnu``
+  * - ``CRAFT_ARCH_BUILD_FOR``
+    - The architecture triplet of the build target. For example:
+
+      - ``amd64``
+      - ``riscv64``
+  * - ``CRAFT_ARCH_BUILD_ON``
+    - The architecture triplet of the build target. For example:
+
+      - ``amd64``
+      - ``riscv64``
+  * - ``CRAFT_PARALLEL_BUILD_COUNT``
+    - The maximum number of concurrent build jobs to execute.
+  * - ``CRAFT_PART_NAME``
+    - The name of the part currently being processed.
+  * - ``CRAFT_PART_SRC``
+    - The path to the part source directory. This is where sources are located
+      after the ``PULL`` step.
+  * - ``CRAFT_PART_SRC_WORK``
+    - The path to the part source subdirectory, if any. Defaults to the part
+      source directory.
+  * - ``CRAFT_PART_BUILD``
+    - The path to the part build directory. This is where parts are built during
+      the ``BUILD`` step.
+  * - ``CRAFT_PART_BUILD_WORK``
+    - The path to the part build subdirectory in case of out-of-tree builds.
+      Defaults to the part source directory.
+  * - ``CRAFT_PART_INSTALL``
+    - The path to the part install directory. This is where built artefacts are
+      installed after the ``BUILD`` step.
+  * - ``CRAFT_OVERLAY``
+    - (If overlays are enabled) The path to the part's layer directory during the
+      ``OVERLAY`` step.
+  * - ``CRAFT_STAGE``
+    - The path to the project's staging directory. This is where installed
+      artefacts are migrated during the ``STAGE`` step.
+  * - ``CRAFT_PRIME``
+    - The path to the final primed payload directory after the ``PRIME`` step.
+
+The following environment variables are also included, but are deprecated:
+
+.. list-table::
+  :header-rows: 1
+
+  * - Variable Name
+    - Description
+  * - ``CRAFT_ARCH_TRIPLET``
+    - The machine-vendor-os platform triplet definition.
+      Use ``CRAFT_ARCH_TRIPLET_BUILD_ON`` or ``CRAFT_ARCH_TRIPLET_BUILD_FOR`` instead.
+  * - ``CRAFT_TARGET_ARCH``
+    - The architecture we're building for. Use ``CRAFT_ARCH_BUILD_FOR`` instead.
+
+Some standard environment variables are also modified during parts execution steps.
+
+``PATH``
+~~~~~~~~
+
+Several paths are prepended to ``PATH`` during step execution, allowing staged
+executables from previous parts as well as already-built executables from the current
+path to be executed without calling their full path. The paths are only added to
+``PATH`` if they exist. These paths are:
```

Comment:
Think there is any value in mentioning the order for this and the sections below?

```suggestion
``PATH`` if they exist. These paths are, in order:
```